NOTES.WIN

```diff
@@ -140,6 +140,35 @@
  OpenSSL BIO layer and your compiler run-time. See the OPENSSL_Applink
  manual page for further details.
 
+ A note on linking with OpenSSL .DLLs
+ ------------------------------------
+
+ When building the OpenSSL .DLLs, import libraries are built alongside
+ them.  These import libraries are named differently for different
+ config targets:
+
+ VC-*           libcrypto.lib, libssl.lib
+ Cygwin*        libcrypto.dll.a, libssl.dll.a
+ mingw*         libcrypto.dll.a, libssl.dll.a
+
+ Unfortunately for the VC-* config targets, this means that static
+ libraries can't be produced alongside DLLs, as they would have the
+ same name as the import libraries.  This is a source of confusion, as
```

Comment:
Formally speaking this is not true. In sense that there is no requirement that  static library actually has to have same name as import in VC case. There even were suggestions to embed VC /Mx flags into library names, which suggests that people would actually appreciate different names for static and import libraries. I probably should clarify that I for one am not fond of idea of embedding the /Mx flag into the name. Things were arranged in such way that it's actually possible to produce single sent of of object modules, and then link them into both .dll (naturally with accompanying import library) and static .lib (naturally with name different from import library) And then compile application with *own* flags and link with either dynamic or static library. For example if developer wants to debug own application it should be sufficient to compile application code with debug flag, one shouldn't have to compile OpenSSL with debug flag. And intention was that it would actually be possible. And it is actually possible.... (Well, modulo that small subset of modules are better be compiled differently for dynamic and static libraries, as discussed elsewhere.)

Well, one can wonder what does this remark have to do with question at hand. I'd say that that the suggestion is another case of projecting of too much of Unix experience. It's supposed to address some confusion, but I'd say that there is confusion about what is the confusion... mingw and cygwin are doing pretty good job arranging things the Unix way and are not confusing at all, not to their target audience.. But as far as Windows audience go, everything would be confusing to some degree. Because there is no **the** Windows way to solve it, everybody would have own idea. So what can one do? One can provide extra flexibility that would allow users to express their ideas about how it should look like more freely (there were some suggestions earlier, I made one too), or one can insist on something more unified (along the line of suggestion for Unix to have separately named static library). But latter case should offer something that would be considered worthy to impose. I mean since everybody has own idea, then suggested idea should distinguish itself by something that would be hard to say no to. For example not demanding that library and application has to be compiled with same flags... But it might be just me...

Reply:
So what I hear is that this section should start with an explanation on why we use the names we do.  A stated goal if you wish.  That would make sense...

Reply:
Yeah... I would also say that one shouldn't mention cygwin at all. This is because, as already said, things are not confusing there at all. One can even wonder if one has to mention mingw... Either way one doesn't have to say that "static libraries can't be produced alongside DLL", it should be sufficient to say user has to recognize that [at least currently!] OpenSSL VC-\* build procedure produces .lib file which is either import library or static one, depending on [no-]shared flag at config time. As for command line examples, more specifically VC-\* one. Examples imply certain responsibility, they are viewed as maintained, which in turn means that you ought to describe under which circumstances they work. For the suggested LINK command to work .lib-s have to reside in current working directory or directory has to be listed in %LIB% environment variable. If you don't say something like that, then you'd probably be better off not providing examples.

BTW, is there specific trigger for this?

Reply:
> BTW, is there specific trigger for this?

Yes, there's been a few times lately when some mistook the import library for an independent static library